pkg/api/v1/types.go

```diff
@@ -951,6 +985,20 @@ type VirtualMachineStatus struct {
 	Ready bool `json:"ready,omitempty"`
 	// Hold the state information of the VirtualMachine and its VirtualMachineInstance
 	Conditions []VirtualMachineCondition `json:"conditions,omitempty" optional:"true"`
+	// StateChangeRequest indicates if the current VMI should be stopped or (re)started
+	StateChangeRequest *VirtualMachineStateChangeRequest `json:"stateChangeRequest,omitempty" optional:"true"`
+}
+
+type VirtualMachineStateChangeRequest struct {
+	// Indicates the UUID of an already-running Virtual Machine Instance that should be stopped.
+	// Stopping an existing VMI will take place before startVMI,
+	// so setting both fields requests a restart.
+	StopVMI *types.UID `json:"stopVMI,omitempty" optional:"true" protobuf:"bytes,5,opt,name=uid,casttype=k8s.io/kubernetes/pkg/types.UID"`
```

Comment:
When looking about the direction all this is going, I am not sure if we should express the requests like this. It might make sense to have an array here which looks somewhat like this:

```yaml
stateChangeRequests: 
  - action: Stop
    uid: 1234
  - action: Start
```

It would allow us also to add more details later. Things like backoff retries, timeouts, ...
Also additional actions like `pause` and `unpause` or poer-off mechanism (guest agent, acpi, pull the plug) can be filled in.

In general we should have a clear decision under which circumstances we want to guarantee something, since VMIs can always be deleted behind our back. The controller could be started, starts a VMI, has an issue and can't clear it's request status because of an API error. In the meantime the VMI gets deleted. What do we want to do now?

Reply:
I like that suggestion. Code now follows what you described here.